src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -70,6 +74,9 @@ public async Task RunAsync(ActionRunStage stage)
                     // Set GITHUB_ACTION_PATH
                     step.ExecutionContext.SetGitHubContext("action_path", ActionDirectory);
 
+                    // Add Expression Functions
+                    step.ExecutionContext.ExpressionFunctions.Add(new FunctionInfo<HashFilesFunction>(PipelineTemplateConstants.HashFiles, 1, byte.MaxValue));
```

Comment:
Let's move this into `RunStepsAsync` because that function should resemble StepsRunner as much as possible

Reply:
So in between these two lines:

```
                Trace.Info($"Processing embedded step: DisplayName='{step.DisplayName}'");

                // Initialize env context
                Trace.Info("Initialize Env context for embedded step");
```